torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4854,6 +4871,8 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
                    dtypes=all_types_and_complex_and(torch.bool, torch.bfloat16),
                    dtypesIfCPU=all_types_and_complex_and(torch.bool, torch.bfloat16),
                    dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
+                   backward_dtypesIfCPU=all_types_and(torch.bool, torch.bfloat16),
```

Comment:
This is interesting. How do you suggest we think of the backward dtypes vs. the "supports_complex_autograd" metadata? Would you recommend removing the latter? 

cc @anjali411

Reply:
`supports_complex_autograd` is not only used on `test_supported_backward` but also used in other test in test_ops.py such as test_variant_consistency_eager, test_variant_consistency_jit.

Reply:
sure, but that seems solvable

we could even add op.supports_autograd(device, dtype) -> bool as a function and replace supports_complex_autograd uses with it, I bet

I just wonder if there should be a "single source of truth" for whether an op supports complex autograd, or maybe there's an advantage to having both?

Reply:
q: is there a chance when an `op` supports complex_autograd, but not backward?

I think this is equivalent to ask if this claim is correct: "An (`op` x `dtype`) combo to support backward **if and only if** `op` support autograd for that `dtype`. if so we can treat them as equal and replace all support_complex_autograd with backward_dtype

Reply:
> q: is there a chance when an `op` supports complex_autograd, but not backward?

I don't understand what you mean. afaik both those things mean the same thing

Reply:
I like the idea of going exclusively to whether the op supports a dtype in backward vs. complex autograd, I'd also like to make a helper function, like `op.supports_autograd(device, dtype)`, to replace the existing queries to supports_complex_autograd

@anjali411, what would you think of that approach? It's more general, but it also loses the readable `supports_complex_autograd` field. Maybe we could add comments above supported backward dtypes describing complex autograd support to make it more readable?

Reply:
+1 on the helper function approach

Reply:
For `pow`, backward is not supported for `float16` on CPU. It looks like this PR might help in such cases, and would also make it easier to document the dtypes an op supports autograd for.

Reply:
I've added a follow up TODO to the description. it seems like a pretty sizable change so will follow up with another PR.

Reply:
I agree with Mike that it would reduce the readability a little to remove `supports_complex_autograd` kwarg, but at the same time it's also bad to have two kwargs sending the same signal. I think it's ok to get rid of the `supports_complex_autograd`, if we create an issue where the forward is supported for complex, but the backward is not and link it in a comment wherever complex autograd is skipped for only backward.

Reply:
@walterddr This PR looks very good, what would you think of extending it to eliminate `supports_complex_autograd` and file the follow-up issues @anjali411 mentioned?

I realize that might be a bit of a pain, but I think it would make the PR more complete. Also there are only six opinfos that current use supports_complex_autograd, which shouldn't be too bad.

Reply:
just made another change to convert `supports_complex_autograd(device_type)` as a function depends on backward dtypes. could you please kindly take another look @mruberry